engine/src/render/lights/plugin.cpp

```diff
@@ -7,8 +9,39 @@
 void cubos::engine::lightsPlugin(Cubos& cubos)
 {
     cubos.resource<RenderEnvironment>();
+    cubos.depends(activePlugin);
 
     cubos.component<DirectionalLight>();
     cubos.component<PointLight>();
     cubos.component<SpotLight>();
+
+    cubos.observer("add active component on add DirectionalLight")
+        .onAdd<DirectionalLight>()
+        .without<ActiveComponent>()
+        .call([](Commands cmds, Query<Entity> query) {
+            for (auto [ent] : query)
+            {
+                cmds.add(ent, ActiveComponent{});
+            }
+        });
+
+    cubos.observer("add active component on add PointLight")
+        .onAdd<PointLight>()
+        .without<ActiveComponent>()
+        .call([](Commands cmds, Query<Entity> query) {
+            for (auto [ent] : query)
+            {
+                cmds.add(ent, ActiveComponent{});
+            }
+        });
+
+    cubos.observer("add active component on add SpotLight")
+        .onAdd<SpotLight>()
+        .without<ActiveComponent>()
+        .call([](Commands cmds, Query<Entity> query) {
+            for (auto [ent] : query)
+            {
+                cmds.add(ent, ActiveComponent{});
+            }
+        });
```

Comment:
Btw if you add this you will also need to add an if to skip over these lights where they're used.
Otherwise, it won't do anything. It's a good idea to ask the graphics team about it.